float64 min_distance
